interfaces/builtin/account_control.go

```diff
@@ -0,0 +1,88 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const accountControlConnectedPlugAppArmor = `
+# Allow creating, modifying and deleting non-system users and account details.
+/{,usr/}bin/passwd ixr,
+/{,usr/}bin/gpasswd ixr,
+/{,usr/}bin/chage ixr,
+/{,usr/}bin/chfn ixr,
+/{,usr/}bin/chsh ixr,
+/{,usr/}bin/expiry ixr,
+/{,usr/}sbin/unix_chkpwd ixr,
+
+# Also allow the batch-mode variants
+/{,usr/}sbin/chpasswd ixr,
+/{,usr/}sbin/chgpasswd ixr,
+/{,usr/}sbin/newusers ixr,
+
+# Allow checking integrity of files
+/{,usr/}sbin/pwck ixr,
+/{,usr/}sbin/grpck ixr,
+
+# Allow adding, removing and modifying users and groups
+/{,usr/}sbin/{add,del}user ixr,
+/{,usr/}sbin/user{add,mod,del} ixr,
+/{,usr/}sbin/{add,del}group ixr,
+/{,usr/}sbin/group{add,mod,del} ixr,
+
+# Only allow modifying the non-system extrausers database
+/var/lib/extrausers/{,**} rwkl,
+
+# Needed by useradd
+/etc/login.defs r,
+/etc/default/useradd r,
+/etc/default/nss r,
+/etc/pam.d/{,*} r,
+
+# Useradd needs netlink
+network netlink raw,
+
+# Capabilities needed by useradd
+capability audit_write,
+capability chown,
+
+# useradd write the result in the log
+/var/log/lastlog rw,
+/var/log/faillog rw,
+`
+
+// Needed because useradd uses a netlink socket
+const accountControlConnectedPlugSecComp = `
+sendto
+recvfrom
+fchown
```

Comment:
stracing useradd we see that it chowns to root:root and to root:shadow. As of recent commits we already allow chowning to root:root, and on Ubuntu shadow is '42', so please adjust to:
```
# useradd requires chowning to 'shadow'
# TODO: dynamically determine the shadow gid to support alternate cores
fchown - 0 42
fchown32 - 0 42
```

I'm actively working on seccomp arg filtering policy (assuming I'm able to get to it with all the reviews! :) and will implement this 'TODO' before the next snapd release.